fdbserver/TestTLogServer.actor.cpp

```diff
@@ -200,21 +201,37 @@ ACTOR Future<Void> TLogTestContext::sendPushMessages(TLogTestContext* pTLogTestC
 		MutationRef m(MutationRef::Type::SetValue, key, val);
 
 		// build commit request
-		LogPushData toCommit(pTLogTestContext->ls, pTLogTestContext->numLogServers /* tLogCount */);
+		LogPushData toCommit(pTLogTestContext->ls, pTLogTestContext->numLogServers);
 		// UID spanID = deterministicRandom()->randomUniqueID();
 		toCommit.addTransactionInfo(SpanContext());
 
-		// for each tag
-		for (uint32_t tagID = 0; tagID < pTLogTestContext->numTagsPerServer; tagID++) {
-			Tag tag(pTLogTestContext->tagLocality, tagID);
-			std::vector<Tag> tags = { tag };
-			toCommit.addTags(tags);
-			toCommit.writeTypedMessage(m, false, true);
+		std::unordered_map<uint16_t, Version> tpcvMap;
+
+		uint32_t replicaCount = 2; // SOMEDAY: paramaterize
+		uint32_t startLogServer = 0;
+		uint32_t teamSize = pTLogTestContext->numLogServers / replicaCount;
+		for (uint32_t tagID = 0; tagID < pTLogTestContext->numTags; tagID++, startLogServer++) {
+			for (uint32_t logServer = startLogServer; logServer < pTLogTestContext->numLogServers;
+			     logServer += teamSize) {
```

Comment:
Why add `teamSize`? will `logServer++` be correct?

Should the loop be `logServer < min(starLogServer + teamSize, pTLogTestContext->numLogServers)`?

Reply:
I add `teamSize` on purpose, because I want to distribute tags over the `tLogs`. So if there are four tTlogs and two tags (replica 2), you would get:

```
TL1 Tag1
TL2 Tag2
TL3 Tag1
TL4 Tag2
```

This is so I can test advancing versions at different rates in version vector. For example, Tag1 could get 5 commits, and Tag2 could get 10. TL2 and TL4 would be further ahead than TL1 and TL3 in version vector.